frontend/src/tests/unit/Api.test.js

```diff
@@ -0,0 +1,51 @@
+import { convertSecondsToStringInfo, getRemainingDate, parseJwt } from '../../utils/Api'
+import { testEndDates } from './testHelpers/endDates'
+import { testInvalidTokens, testProfessorTokens, testStudentTokens } from './testHelpers/tokens'
+
+describe('parseJwt() tests', () => {
+  it.each(testStudentTokens)('student token %s parsed correctly', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toEqual(token.token_data)
+  })
+  it.each(testProfessorTokens)('professor token %s parsed correctly', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toEqual(token.token_data)
+  })
+  it.each(testInvalidTokens)('invalid token %s throws error', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toBeNull()
+  })
+})
+
+describe('getRemainingDate() tests', () => {
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date('2022-05-13T12:00:00.000Z'))
+  })
+
+  it.each(testEndDates)('returns correct remaining time to date %s', (endDate) => {
+    //when
+    const remainingDate = getRemainingDate(endDate.date)
+
+    //then
+    expect(remainingDate.asDays()).toEqual(endDate.remainingDays)
+  })
+})
+
+describe('convertSecondsToStringInfo() tests', () => {
+  it.each(testEndDates)('returns correct date from seconds %s', (endDate) => {
```

Comment:
Nazwa testu wskazuje, że podamy datę w sekundach co się nie dzieje bo podajemy normalną datę w stringu.
W ogóle nazwa testowanej funkcji sugeruje, że podawana `endDate` będzie w sekundach. W praktyce podajemy tam raczej milisekundy. 
Podsumowując:
1. W tym teście zmienił bym opis sugerując, że sprawdzamy dla daty ze stringa 
2. Dodałbym test sprawdzający to samo, ale dla `endDate` w milisekundach
3. Nieco przerobił bym samą testowaną funkcję tak żeby dla `nulla`, `undefined`, błędnego stringa i innych głupot (takich że `getRemainingDate()` sobie nie radzi) dawał inny string.

Reply:
ewentualnie 4. Zmieniłbym nazwę testowanej funkcji - `...Seconds...` -> `...Miliseconds...` bo takie jest praktyczne zastosowanie

Reply:
Sprawdziłem - 2 nie wydaje mi się być potrzebne, sprawdziłem kod i jedyne wywołania tej funkcji są już dla obiektu daty

Reply:
Odnośnie 4 -> prędzej `convertDateToStringInfo`

Reply:
do 3 -> można sprawdzać czy jest przekazany obiekt `Date`, tak zrobię

Reply:
Faktycznie krok 2 zbędny bo zniknął case z wykorzystaniem ms. Dzięki ;)